### PR TITLE
CA-213568: Restore VM power state on checkpoint failure.

### PR DESCRIPTION
**Pull request by sharady**
Restore VM power state before raising an exception for
checkpoint failure.

Signed-off-by: Sharad Yadav <sharad.yadav@citrix.com>

### CODE REVIEW

**Comment by simonjbeaumont** [comment]
@xen-git retest this please.

**Comment by simonjbeaumont** [comment]
I can't see anywhere in this pull request that does anything with the VM power state. Maybe I'm missing something. Unless I'm missing something obvious, please can you elaborate on how the changes do what you say in the commit message?

File: ocaml/xapi/xapi_vm_snapshot.ml [hunk 1]
```diff
@@ -203,12 +203,13 @@ let checkpoint ~__context ~vm ~new_name =
 		end;
 
 		(* snapshot the disks and the suspend VDI *)
-		let snap =
+		let snap, err =
 			if not (TaskHelper.is_cancelling ~__context) then begin
-				try Some (Xapi_vm_clone.clone Xapi_vm_clone.Disk_op_checkpoint ~__context ~vm ~new_name ~snapshot_info_record:!snapshot_info)
```
Comment (sharady) [inline ocaml/xapi/xapi_vm_snapshot.ml]: As per CA-213568: Call `Xapi_vm_clone.clone Xapi_vm_clone.Disk_op_checkpoint ~__context ~vm ~new_name ~snapshot_info_record:!snapshot_info` fails with error `Server_error(SR_BACKEND_FAILURE_44, [ ; There is insufficient space;  ])`. This exception is raised before calling subsequent call `Xapi_xenops.resume ~__context ~self:vm ~start_paused:false ~force:false`. This is causing the VM to be in `suspended` state during a checkpoint.

Comment (simonjbeaumont) [inline ocaml/xapi/xapi_vm_snapshot.ml]: Ah ha! I see now. The call to `Xapi_xenops.resume` was not in the context of the patch. Only visible when expanding the file.

**Pull request merged**